src/index.tsx

```diff
@@ -361,6 +395,9 @@ interface UseValidatedInputOpts<T extends FormValidations> {
   formValidations?: T;
   errorMessages?: ErrorMessages;
   serverFormInfo?: ServerFormInfo<T>;
+  ref?:
+    | React.ForwardedRef<HTMLInputElement | null | undefined>
+    | React.Ref<HTMLInputElement | null | undefined>;
```

Comment:
I think the ergonomics might be better if we took this as an option to `getInputAttrs` instead, since it would be assigned to the `input` element.  That aligns with the other input attributes like `id`, `className`, etc.  `name` is the one-off that goes into `useValidatedInput` since it's not just an attribute, but also the lookup for validations and error messages

```jsx
// ref and id separate
function EmailAddress() {
  let ref = useRef();
  let { getInputAttrs } = useValidatedInput({ name: "emailAddress", ref });

  return <input {...getInputAttrs({ id: 'email' })} />;
}

// ref and id colocated
function EmailAddress() {
  let ref = useRef();
  let { getInputAttrs } = useValidatedInput({ name: "emailAddress" });

  return <input {...getInputAttrs({ id: 'email', ref })} />;
}
```

Reply:
Tested though and this works likes a charm - thanks!  I really need to wire up some unit testing in here 😬

Reply:
Hm, does accepting through `getInputAttrs` instead of the hook pose an issue with the `useCallback` stuff going on in there since it could then be optionally called?  That's no good...

Reply:
I took a stab at this in [e0300ef](https://github.com/brophdawg11/remix-validity-state/pull/19/commits/e0300efeb0a9d2cc4683cf5db0f763d8df1ccc73) - does that seem like it would work?  Wasn't 100% sure if that type of ref assignment inside a non-hook function was ok - but it does seem to work when I tested in the demo app.

If there's fundamental issues with doing this in `getInputAttrs` then we can revert my commit 😄

Reply:
Reverted due to "don't assign refs in render" concerns